regression/cbmc/sat-solver-warning/test.desc

```diff
@@ -0,0 +1,9 @@
+CORE broken-z3-smt-backend broken-cprover-smt-backend 
+test.c
+--sat-solver lingeling
+^EXIT=0$
+^SIGNAL=0$
+The specified solver, 'lingeling', is not available\. The default solver will be used instead\.
+\[main\.assertion\.1] line \d+ assertion x == y: SUCCESS
+^VERIFICATION SUCCESSFUL$
+--
```

Comment:
This seems slightly brittle - what happens if someone adds `lingeling` to the list of built in solvers like CaDiCaL?

Reply:
Well, changing this one test seems easy enough? One option would be to extend the code that does the checks to also emit a warning if a completely different solver string is passed (none of the solvers that the code base knows about). Any other suggestions on how to avoid churn here @TGWDB?

Reply:
That seems like a good path forward.

Reply:
The PR already emits an error if the solver is none of the ones it knows about: https://github.com/diffblue/cbmc/pull/7493/files#diff-d9c9d73e84f06014b049a86daaa54f9be9744f6a3d998ac4acdc9d88989bf364R332

Anything else I need to do here?

Reply:
I was concerned that the tested behaviour is for lingeling being missing. Can we test with a solver name that is not existent and match that failure state? (The code is there, this is just updating the test.)

Reply:
I added a new test (`cbmc/sat-solver-error`) that checks for the case of a non-existent solver and that cbmc exits with an error.